daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -85,3 +94,137 @@ func (l *JSONFileLogger) Close() error {
 func (l *JSONFileLogger) Name() string {
 	return Name
 }
+
+func decodeLogLine(dec *json.Decoder, l *jsonlog.JSONLog) (msg logger.Message, err error) {
+	l.Reset()
+	if err = dec.Decode(l); err != nil {
+		return
+	}
+	msg.Source = l.Stream
+	msg.Timestamp = l.Created
+	msg.Line = []byte(l.Log)
+	return
+}
+
+// Reads from the log file
+func (l *JSONFileLogger) ReadLogs(config logger.ReadConfig) *logger.LogWatcher {
+	logWatcher := logger.NewLogWatcher()
+
+	go l.readLogs(logWatcher, config)
+	return logWatcher
+}
+
+func (l *JSONFileLogger) readLogs(logWatcher *logger.LogWatcher, config logger.ReadConfig) {
+	defer func() {
```

Comment:
Nope, was doing other stuff, in here, forgot to remove it when I got rid of the other stuff.